actions/actions.py

```diff
@@ -234,26 +234,34 @@ def validate_location(
                 "fields": "formatted_address,geometry",
             }
         )
-        response = requests.get(
-            (
-                f"https://maps.googleapis.com"
-                f"/maps/api/place/findplacefromtext/json?{querystring}"
-            )
-        )
-        location = response.json()
-        if location["candidates"]:
-            formatted_address = location["candidates"][0]["formatted_address"]
-            geometry = location["candidates"][0]["geometry"]["location"]
-            latitude = geometry["lat"]
-            longitude = geometry["lng"]
-            return {
-                "location": formatted_address,
-                "latitude": latitude,
-                "longitude": longitude,
-            }
+
+        if hasattr(httpx, "AsyncClient"):
+            # from httpx>=0.11.0, the async client is a different class
+            HTTPXClient = getattr(httpx, "AsyncClient")
         else:
-            dispatcher.utter_message(template="utter_incorrect_location")
-            return {"location": None}
+            HTTPXClient = getattr(httpx, "Client")
```

Comment:
Ah, this is an interesting way to deal with the versions.

I just want to check if the client API is the same between the two versions?

If so we can also apply this logic to the connector.

Looking at the dependancy tree, Rasa depends on `sanic` (version pinned), and `sanic` depends on `httpx` (also version pinned), so the only time we should have an issue is when upgrading our Rasa version.

Reply:
Yup, that's what I figured. I initially put a "TODO" but that felt like it would be a comment that future me would not be happy to see..